web/src/client/questions.js

```diff
@@ -0,0 +1,161 @@
+ */
+
+const DBUS_CONFIG = {
+  questions: {
+    path: "/org/opensuse/DInstaller/Questions1",
+    ifaces: {
+      objectManager: "org.freedesktop.DBus.ObjectManager"
+    }
+  },
+  question: {
+    ifaces: {
+      question: "org.opensuse.DInstaller.Question1",
+      luksActivation: "org.opensuse.DInstaller.Question.LuksActivation1"
+    }
+  }
+};
+
+const QUESTION_TYPES = {
+  generic: "generic",
+  luksActivation: "luksActivation"
+};
+
+/**
+ * Builds a question from the given D-Bus question
+ *
+ * @param {Object} dbusQuestion
+ * @returns {Object}
+*/
+function buildQuestion(dbusQuestion) {
+  const getIfacesAndProperties = (dbusQuestion) => Object.values(dbusQuestion)[0];
+
+  const getIfaces = (dbusQuestion) => Object.keys(getIfacesAndProperties(dbusQuestion));
+
+  const fetchValue = (ifaceProperties, key) => {
+    const dbusValue = ifaceProperties[key];
+
+    if (dbusValue) return dbusValue.v;
+
+    return null;
+  };
+
+  const question = {};
+  const ifaces = getIfaces(dbusQuestion);
+  const ifacesAndProperties = getIfacesAndProperties(dbusQuestion);
+
+  if (ifaces.includes(DBUS_CONFIG.question.ifaces.question)) {
+    const dbusProperties = ifacesAndProperties[DBUS_CONFIG.question.ifaces.question];
+
+    question.type = QUESTION_TYPES.generic;
+    question.id = fetchValue(dbusProperties, "Id");
+    question.options = fetchValue(dbusProperties, "Options");
+    question.defaultOption = fetchValue(dbusProperties, "DefaultOption");
+    question.text = fetchValue(dbusProperties, "Text");
+    question.answer = fetchValue(dbusProperties, "Answer");
+  }
+
+  if (ifaces.includes(DBUS_CONFIG.question.ifaces.luksActivation)) {
+    const dbusProperties = ifacesAndProperties[DBUS_CONFIG.question.ifaces.luksActivation];
+
+    question.type = QUESTION_TYPES.luksActivation;
+    question.password = fetchValue(dbusProperties, "Password");
+    question.attempt = fetchValue(dbusProperties, "Attempt");
+  }
+
+  return question;
+}
+
+export default class QuestionsClient {
+  constructor(dbusClient) {
+    this._client = dbusClient;
+  }
+
+  /**
+   * Return all the questions
+   *
+   * @return {Promise.<Array.<Object>>}
+   */
+  async getQuestions() {
+    const dbusQuestions = await this._client.call(
+      DBUS_CONFIG.questions.path,
+      DBUS_CONFIG.questions.ifaces.objectManager,
+      "GetManagedObjects",
+      null
+    );
+
+    // Note: dbusQuestions contains an empty object when there are no questions.
+    return dbusQuestions.filter(q => Object.keys(q).length !== 0).map(buildQuestion);
+  }
+
+  /**
+   * Answer with the information in the given question
+   *
+   * @param {Object} question
+   */
+  async answer(question) {
+    const path = DBUS_CONFIG.questions.path + "/" + question.id;
+
+    if (question.type === QUESTION_TYPES.luksActivation) {
+      const proxy = this._client.proxy(DBUS_CONFIG.question.ifaces.luksActivation, path);
+      await proxy.wait();
```

Comment:
I would expect `proxy.wait` to be [already called](https://github.com/yast/d-installer/pull/150/files#diff-2d2ba2cb24f885ee0c6fab7c64735a70ecb224aec2c08bf55aa0b88be2c4f6aaR37). But perhaps I lost track...

Reply:
Note that questions are dinamically generated and removed. We need to create a new proxy everytime we are dealing we a new question.

Reply:
Oh, my fault, you are doing it right.